### PR TITLE
TT-9934 Add flag for hybrid analytics enable or disable

### PR DESCRIPTION
**Pull request by mitjaziv**
## Description

```
Change adds flag to enable or disable analytics sending for Hybrid Gateway.
```

## Motivation and Context

With current hybrid helm chart implementation, if you don't want to use Pump to send
analytics and if you disable Pump in value, analytics on hybrid gateway are disabled. 

This change enable you to specifically turn analytics on hybrid gateway on or off.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] Documentation updates or improvements.

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [x] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
     - [ ] I have manually updated the README(s)/documentation accordingly.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by caroltyk** [comment]
@mitjaziv Can you update the PR to use below default values for the new option?

tyk-data-plane: true
tyk-oss: false
tyk-gateway: false
tyk-stack: true

File: components/tyk-gateway/templates/deployment-gw-repset.yaml [hunk 1]
```diff
@@ -155,6 +155,8 @@ spec:
             value: "file"
         {{- end}}
         {{- if .Values.global.remoteControlPlane.enabled }}
+          - name: TYK_GW_ENABLEANALYTICS
+            value: "{{.Values.gateway.analyticsEnabled}}"
```
Comment (caroltyk) [inline components/tyk-gateway/templates/deployment-gw-repset.yaml:159]: should this be applicable for all but not only when remoteControlPlane is enabled?

File: components/tyk-gateway/values.yaml [hunk 2]
```diff
@@ -249,6 +249,9 @@ gateway:
   ##     readOnly: true
   extraVolumeMounts: []
 
+  # analyticsEnabled property is used to enable or disable sending analytics to Control Plane.
```
Comment (komalsukhani) [inline components/tyk-gateway/values.yaml:252]: `TYK_GW_ENABLEANALYTICS` controls if the analytics will be recorded or not. Reference to docs https://tyk.io/docs/tyk-oss-gateway/configuration/#enable_analytics

Can you update the comment accordingly?

**Review by caroltyk — APPROVED** [review]
LGTM

**Comment by caroltyk** [comment]
@jakub-bochenski did you add the new config `analyticsEnabled: "true"`?

**Comment by jakub-bochenski** [comment]
I did, but that's what a "breaking change" is - I had to update my config to restore the way it was working in previous version

**Comment by joshblakeley** [comment]
@caroltyk agree with Jakub that is really a bug on the charts.

**Comment by caroltyk** [comment]
@jakub-bochenski @joshblakeley we're checking this issue. Which chart are you using @jakub-bochenski ?

**Comment by jakub-bochenski** [comment]
I'm using the tyk-gateway chart directly (and the tyk-pump one, but that's not relevant here)

In my case the exact change was:
 - before it was unspecified
 - now it's disabled

As seen in this fragment of helm diff output:
Screenshot_20230914_175917

So before the change I had analytics enabled by means of `tyk.conf` file.
After the change analytics is disabled by default via environment (overriding the `tyk.conf` file)

**Comment by komalsukhani** [comment]
@jakub-bochenski I have tried to make the solution backward compatible in PR #123 
Any feedback or comments much appreciated.

**Pull request merged**